src/v/cluster/data_migration_backend.cc

```diff
@@ -564,11 +587,13 @@ ss::future<errc> backend::do_topic_work(
     }
 }
 
-void backend::abort_all_topic_work() {
+ss::future<> backend::abort_all_topic_work() {
     for (auto& [nt, tsws] : _active_topic_work_states) {
-        tsws->rcn.request_abort();
+        tsws->rcn().request_abort();
+    }
+    while (!_active_topic_work_states.empty()) {
+        co_await _active_topic_work_states.begin()->second->future();
```

Comment:
this code is pretty confusing, can we simply `co_await` for the abort future in the first loop ?

Reply:
It think it works quicker with separate loops. If one of the works is slow to stop we signal to others to stop earlier.